db/migrate/20240712094833_create_academic_years.rb

```diff
@@ -0,0 +1,25 @@
+class CreateAcademicYears < ActiveRecord::Migration[7.1]
+  class AcademicYearForMigration < ApplicationRecord
+    self.table_name = :academic_years
+  end
+
+  def change
+    create_table :academic_years, id: :uuid do |t|
+      t.string :name
+      t.date :starts_on
+      t.date :ends_on
+
+      t.timestamps
+    end
+
+    up_only do
+      [2023, 2024, 2025].each do |year|
+        AcademicYearForMigration.create!(
+          name: "#{year} to #{year + 1}",
+          starts_on: Date.new(year, 8, 1),
+          ends_on: Date.new(year + 1, 7, 31),
+        )
+      end
+    end
```

Comment:
This belongs in a data migration.

Reply:
Does it need to be? This is a safe migration.

Reply:
The idea of using a data migration is to clearly separate changes to the database, with data entry.
This is so if need be, the data entry can be reverted without the need to roll back the changes to the database as well.

I would also argue that both these migrations run `up_only` with no way of reverting the data entry is these are incorrect. They will also run again if the migration is reverted, creating additional/duplicate AcademicYears and ClaimWindow records. 

So is it a safe migration?

Reply:
If the migration is reverted, the tables are dropped, along with the data. There will be no duplicate entries.

Reply:
But the issue is that there is likely nothing wrong with the database part of the migration, only the data entry. So rather than reverting the whole thing, only revert the thing that's incorrect.

Reply:
It's true that we do have a pattern for data migrations, which was established in https://github.com/DFE-Digital/itt-mentor-services/pull/659. Changes to the database schema should go in `db/migrate` whereas data migrations should go in `db/data`.

It would be cleaner not to create or update records in migrations. None of the existing schema migrations currently perform data changes.

For the sake of consistency with existing migrations, would you mind moving it over @Nitemaeric? 🙏🏻 

It should be relatively easy with this generator:

```
bundle exec rails generate data_migration NAME
```